src/wallet/spend.cpp

```diff
@@ -575,11 +575,11 @@ std::optional<SelectionResult> SelectCoins(const CWallet& wallet, CoinsResult& a
     // Deduct preset inputs amount from the search target
     CAmount selection_target = nTargetValue - pre_set_inputs.total_amount;
 
-    // If automatic coin selection was disabled, we just want to return the preset inputs result
-    if (!coin_control.m_allow_other_inputs) {
-        // 'selection_target' is computed on `PreSelectedInputs::Insert` which decides whether to use the effective value
-        // or the raw output value based on the 'subtract_fee_outputs' flag.
-        if (selection_target > 0) return std::nullopt;
+    // Return if automatic coin selection is disabled, and we don't cover the selection target
+    if (!coin_control.m_allow_other_inputs && selection_target > 0) return std::nullopt;
+
+    // Return if we can cover the target only with the preset inputs
+    if (selection_target <= 0) {
         SelectionResult result(nTargetValue, SelectionAlgorithm::MANUAL);
```

Comment:
nit: I'd suggest to create `SelectionResult` for preselected inputs unconditionally. This way use can reuse it later when you need to merge it with automatic coin selection.

Reply:
I thought about it but It's not that straightforward. The `SelectionResult` target is provided on the constructor and then only modified by the `Merge` function (which adds the merged selection target to the base one).

So, it's not clear with which target the selection result should be initialized:

1) If the preset inputs cover the target amount, then it needs to be initialized to the target amount, not to the preset inputs total amount. So the selection result can be returned early.

2) If the preset inputs don't cover the target amount, then it needs to be initialized to the preset inputs total amount, because we need to cover the total amount when we merge it with the `SelectionResult` that comes from `AutomaticCoinSelection` (which contains a selection for the target amount minus the preset inputs total).

Reply:
Yap, agreed.